src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -12,11 +19,48 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Phone create(Phone phone) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(phone);
+            transaction.commit();
+            return phone;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add phone \"" + phone + "\" to DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Phone> query = criteriaBuilder.createQuery(Phone.class);
+            Root<Phone> phoneRoot = query.from(Phone.class);
+            Predicate phonePredicate = criteriaBuilder.and();
+
+            for (Map.Entry<String, String[]> entry : params.entrySet()) {
+                CriteriaBuilder.In<Object> paramsPredicate =
+                        criteriaBuilder.in(phoneRoot.get(entry.getKey()));
+                for (String value : entry.getValue()) {
+                    paramsPredicate.value(value);
+                }
+                phonePredicate = criteriaBuilder.and(phonePredicate, paramsPredicate);
+            }
+
+            query.where(phonePredicate);
+            return session.createQuery(query).getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get phones by params \"" + params
```

Comment:
You can't properly print out map with array inside just calling by name "params".